x/epochs/keeper/epoch_infos.go

```diff
@@ -22,22 +44,49 @@ func (k Keeper) GetEpochInfo(ctx sdk.Context, identifier string) (types.EpochInf
 
 // SetEpochInfo set epoch info
 func (k Keeper) SetEpochInfo(ctx sdk.Context, epoch types.EpochInfo) {
+	k.setEpochDuration(ctx, epoch)
+	k.setEpoch(ctx, epoch)
+}
+
+// SetEpochDuration set epoch duration by identifier
+func (k Keeper) setEpochDuration(ctx sdk.Context, epoch types.EpochInfo) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefixEpochDuration)
+	store.Set([]byte(epoch.Identifier), durationToBz(epoch.Duration))
+}
+
+// SetEpochInfo set epoch duration by identifier
+func (k Keeper) setEpoch(ctx sdk.Context, epoch types.EpochInfo) {
 	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefixEpoch)
 	bz := k.cdc.MustMarshal(&epoch)
-	store.Set([]byte(epoch.Identifier), bz)
+	store.Set(durationToBz(epoch.Duration), bz)
 }
 
 // DeleteEpochInfo delete epoch info
 func (k Keeper) DeleteEpochInfo(ctx sdk.Context, identifier string) {
-	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefixEpoch)
+	duration, found := k.GetEpochDuration(ctx, identifier)
+	if found {
+		k.deleteEpochDuration(ctx, identifier)
+		k.deleteEpoch(ctx, duration)
+	}
+}
+
+// DeleteEpochDuration delete epoch duration by identifier
+func (k Keeper) deleteEpochDuration(ctx sdk.Context, identifier string) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefixEpochDuration)
 	store.Delete([]byte(identifier))
 }
 
-// IterateEpochInfo iterate through epochs
+// DeleteEpoch delete epoch info
+func (k Keeper) deleteEpoch(ctx sdk.Context, duration []byte) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefixEpoch)
+	store.Delete(duration)
+}
+
+// IterateEpochInfo iterate through epochs DESC by duration
 func (k Keeper) IterateEpochInfo(ctx sdk.Context, fn func(index int64, epochInfo types.EpochInfo) (stop bool)) {
 	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefixEpoch)
 
-	iterator := sdk.KVStorePrefixIterator(store, nil)
+	iterator := sdk.KVStoreReversePrefixIterator(store, nil)
```

Comment:
I used descending numerical ordering here, as suggested in https://github.com/tharsis/evmos/issues/540#issue-1221792563.
Should I add this PR under `API Breaking` because of this change? 
There are no other API changes.

Reply:
I reverted to ascending order.

Default ordering should be consistent with the execution order.
Inflation runs daily. Incentives run weekly. Incentives are given from inflation rewards, so inflation rewards need to be calculated first.